File: backend/Concord/Concord.Application/Models/VoiceParticipantSummary.cs
using System.Text.Json.Serialization;

namespace Concord.Application.Models;

public class VoiceParticipantSummary
{
    [JsonPropertyName("UserId")]
    public Guid UserId { get; set; }

    [JsonPropertyName("Identity")]
    public string? Identity { get; set; }

    [JsonPropertyName("JoinedAt")]
    public DateTime JoinedAt { get; set; }
}

File: mobile/concord/lib/api/models/voice_participant_summary.dart
import 'json_utils.dart';

class VoiceParticipantSummary {
  const VoiceParticipantSummary({
    required this.userId,
    required this.identity,
    required this.joinedAt,
  });

  factory VoiceParticipantSummary.fromJson(Map<String, dynamic> json) {
    return VoiceParticipantSummary(
      userId: json.field('UserId') as String,
      identity: json.field('Identity') as String,
      joinedAt: parseDateTime(json.field('JoinedAt')),
    );
  }

  final String userId;
  final String identity;
  final DateTime joinedAt;
}
